spark/src/main/scala/org/apache/spark/sql/comet/operators.scala

```diff
@@ -149,6 +149,9 @@ abstract class CometNativeExec extends CometExec {
   /**
    * The serialized native query plan, optional. This is only defined when the current node is the
    * "boundary" node between native and Spark.
+   *
+   * Note that derived classes of `CometNativeExec` must have `serializedPlanOpt` in last product
```

Comment:
Alternatively, we can use reflection to look for the `serializedPlanOpt` field in the product and only apply the map function on it. With this we don't have to enforce the ordering.

Reply:
Although you can find `serializedPlanOpt` field from all declared fields in the product, you don't know its order in the constructor. We need to have these arguments in the exact order so we can re-construct the product by calling `makeCopy`.

Reply:
Hmm, a bit trouble for `None` as you cannot get inside value for reflection (as it doesn't have that).

Reply:
Hmm why do we need to know the value? I thought we just need to look at the field names one by one

Reply:
Oh, I thought we want to check the type, i.e., Array[Byte]?

Reply:
By name sounds okay. I am occupied by other stuffs. I will try it later.

Reply:
I only thought about checking the name, not the type, which I think is good enough for now. Checking type may not be easy because of type erasure (unless use `TypeTag`?)

Reply:
Hmm, but when we iterate product element, we don't have its name. In Scala 2.13, there is `productElementNames` API, but we still have 2.12 support.

Reply:
Although we can get all its fields by reflection like `getDeclaredFields`, its order is not same as product element order which we need.

Reply:
Yea, seems `getDeclaredFields` doesn't guarantee the order of returned fields. It is a bit dangeous to assume its order is same as product elements.

Reply:
I see, thanks for looking into this! I think the current approach is OK too, as long as we are aware that `serializedPlanOpt` should be declared as the last field.

Reply:
Not sure if [this](https://stackoverflow.com/questions/16079113/scala-2-10-reflection-how-do-i-extract-the-field-values-from-a-case-class-i-e/16079804#16079804) is useful. Scala 2.13 added `productElementNames` however we still need to handle 2.12 ..

Reply:
I don't make reflection approach work. Instead of spending time on it, I use the wrapper case class approach.

Reply:
OK np. Thanks for trying it.